### PR TITLE
chore(upgrade): Upgrade to java 17, Liferay removed, and authorization enhancement

### PR DESCRIPTION
**Pull request by smrutis1**
### Description

1. Liferay removed
2. Java version update to  17
3. Rest authorization mechanism enhancement to support new Oauth2 flows and removed deprecated flows to support new UI 

#### How to reproduce
Steps: 
1. Update the java version to 17  (Tested in `openjdk 17.0.11 2024-04-16`)
2. Download tomcat 10.1.19 [Tomcat 10.1.19](https://ftp.unicamp.br/pub/apache/tomcat/tomcat-10/v10.1.19/bin/apache-tomcat-10.1.19.tar.gz) or above and configure it.
3. Build the packages `mvn clean package -DskipTests` 
4.  Copy wars to tomcat webapp folder `cp $(find . -name "*.war")  /opt/apache-tomcat-10.1.19/webapps/` (TODO: Need to be automated)
5. Create initial client and user by executing the script `scripts/addUnsafeDefaultClient.sh`(Will create the default user = `admin@sw360.org` and password `12345`)
6. start the server
7. Create token for e.g in rest client like Postman as follows.
     - For basic authorization (Not recommended for prod.):
        - [Basic Auth.webm](https://github.com/eclipse-sw360/sw360/assets/46560890/e876fa20-5fd3-4cac-a35b-55ac1dbdf37d)
     - For Oauth2: 
        -  [Oauth2.webm](https://github.com/eclipse-sw360/sw360/assets/46560890/610f4c04-3ef9-459e-8903-5a486ecc44cc)
 8. To test with new UI. kindly deploy it with PR [286](https://github.com/eclipse-sw360/sw360-frontend/pull/286)

Describe the bug and list the steps you used when the issue occurred.

#### Screenshots ( if applicable )

Add screenshots to help explain your problem.

### CODE REVIEW

**Review by heliocastro — CHANGES_REQUESTED** [review]
Please fix license headers
Please split this commit in logical steps:
* One to remove Liferay
* One to the changes configuration to Java 17
* One to the changes on rest

**Comment by ag4ums** [comment]
@smrutis1, as discussed this in the community call last week, please split the PR in different commits,
 

> Please fix license headers Please split this commit in logical steps:
> 
> * One to remove Liferay
> * One to the changes configuration to Java 17
> * One to the changes on rest

**Comment by smrutis1** [comment]
Thanks looking into the PR. I am working on the splitting commits as per the comment. Meanwhile it can be tested along with new UI, I have added/updated the steps in description.

**Comment by heliocastro** [comment]
@smrutis1 I committed the Java 17 changes you requested. I needed to:
- Disable tests for src-licenseinfo, http-support, importers, resource-server, client
- Update thrift to 0.20.0

The rest need have the tests fixed, but you are changing it on this PR

**Comment by heliocastro** [comment]
@smrutis1 Don't forget to rebase your branch

**Comment by smrutis1** [comment]
(from discussion):
- Rebased the branch
- Reset the java 17 related changes for couchdb-lucene and added a patch file i.e.  `scripts/patches/couchdb-lucene_Java17.patch` instead.

File: libraries/importers/pom.xml [hunk 1]
```diff
@@ -25,7 +25,6 @@
 
 	<properties>
 		<artifact.deploy.dir>${liferay.deploy.dir}</artifact.deploy.dir>
-		<maven.test.skip>true</maven.test.skip>
```
Comment (heliocastro) [inline libraries/importers/pom.xml:28]: @smrutis1 revert this change

**Review by heliocastro — CHANGES_REQUESTED** [review]
@smrutis1 You removed the test ignore stage, please revert the mentioned one and rebase again with main, just check if no test disabling are removed

File: libraries/importers/pom.xml (hunk 1 above)
Comment (smrutis1) [reply libraries/importers/pom.xml:28]: `importers` module test case is fixed [ref](https://github.com/eclipse-sw360/sw360/actions/runs/9873727882/job/27270450726?pr=2501#step:12:19940). Hence its removed. Let me know.

Comment (heliocastro) [reply libraries/importers/pom.xml:28]: Yep, but the PR still not passing, should we merge it anyway ?

**Comment by heliocastro** [comment]
Now the authorization-server tests are missing

**Comment by ag4ums** [comment]
Please skip authorization server test cases, 
Once this PR is merged we will work on issue #2510

**Comment by hoangnt2** [comment]
I tested this PR. Everything work as expected

**Comment by smrutis1** [comment]
Skipped the authorization server test case
